app/controllers/lcms/engine/api/base_controller.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module Lcms
+  module Engine
+    module Api
+      class BaseController < ActionController::Base
+        protect_from_forgery with: :null_session
+
+        before_action :authenticate_request
+
+        private
+
+        def authenticate_request
+          timestamp = request.headers['X-Api-Timestamp']
+          signature = request.headers['X-Api-Signature']
+
+          unless timestamp && signature
+            render json: { error: 'Missing authentication headers' }, status: :unauthorized
+            return
+          end
+
+          # Verify timestamp is within 5 minutes
+          if (Time.now.to_i - timestamp.to_i).abs > 300
```

Comment:
better to move `300` to constant

Reply:
and we may need to add `skip_before_action :verify_authenticity_token`

Reply:
Added the constant. skip_before_action :verify_authenticity_token shouldn't be necessary when using protect_from_forgery with: :null_session